target/linux/generic/backport-6.6/735-05-v6.10-net-phy-realtek-configure-SerDes-mode-for-rtl822xb-P.patch

```diff
@@ -0,0 +1,210 @@
+From 53b084c68374b2a165b355ac8c5ba3ff827037d9 Mon Sep 17 00:00:00 2001
+From: Alexander Couzens <lynxis@fe80.eu>
+Date: Tue, 9 Apr 2024 09:30:11 +0200
+Subject: [PATCH 5/9] net: phy: realtek: configure SerDes mode for rtl822xb
+ PHYs
+MIME-Version: 1.0
+Content-Type: text/plain; charset=UTF-8
+Content-Transfer-Encoding: 8bit
+
+The rtl8221b and rtl8226b series support switching SerDes mode between
+2500base-x and sgmii based on the negotiated copper speed.
+
+Configure this switching mode according to SerDes modes supported by
+host.
+
+There is an additional datasheet for RTL8226B/RTL8221B called
+"SERDES MODE SETTING FLOW APPLICATION NOTE" where a sequence is
+described to setup interface and rate adapter mode.
+
+However, there is no documentation about the meaning of registers
+and bits, it's literally just magic numbers and pseudo-code.
+
+Signed-off-by: Alexander Couzens <lynxis@fe80.eu>
+[ refactored, dropped HiSGMII mode and changed commit message ]
+Signed-off-by: Marek Behún <kabel@kernel.org>
+[ changed rtl822x_update_interface() to use vendor register ]
+[ always fill in possible interfaces ]
+[ only apply to rtl8221b and rtl8226b phy's ]
+[ set phydev->rate_matching in .config_init() ]
+Signed-off-by: Eric Woudstra <ericwouds@gmail.com>
+
+Reviewed-by: Russell King (Oracle) <rmk+kernel@armlinux.org.uk>
+Reviewed-by: should come before them, without any blank lines. As the
```

Comment:
Sorry for the noise but wth happen here AHAHAHHA

Reply:
https://lore.kernel.org/netdev/417f3e08-136b-47db-880b-c80aa5698c9b@lunn.ch/

> Hi Eric
>
> The ordering of these tags should put your Signed-off-by last.
> **Reviewed-by: should come before them, without any blank lines. As the**
> patch makes it way towards mainline, each Maintainer will add a
> Signed-off: by, and it is nice to keep them together.

Patchwork saw a line starting with `Reviewed-by:` and "helpfully" merged it into the patch automatically.

Reply:
... and it ended up in Torvald's tree like that https://github.com/torvalds/linux/commit/deb8af5243504e379878ae3f9a091b21422d65b2

Reply:
Oh god... OK I'M DEAD AHAHAHHAHAHAHAH lets pretend nobody notice that